### PR TITLE
update mux and add video subtitle functions

### PR DESCRIPTION
**Pull request by joelhooks**
uploads

### CODE REVIEW

File: src/app/api/mux/route.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * MUX Direct Upload Endpoint that returns a signed upload URL
+ *
+ * @param req
+ */
+export async function POST(req: NextRequest) {
+  try {
+    const response = await fetch(`${baseUrl}/video/v1/uploads`, {
+      headers: {
+        Authorization: `Basic ${Buffer.from(
+          `${process.env.MUX_ACCESS_TOKEN_ID}:${process.env.MUX_SECRET_KEY}`,
```
Comment (joelhooks) [inline src/app/api/mux/route.ts]: need to check/add all env vars

File: src/inngest/functions/mux/add-srt-to-mux-asset.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+    if (videoResource) {
+      const muxAsset = await step.run('get the mux asset', async () => {
+        const assetId = videoResource.muxAssetId
+        const {data} = await fetch(
+          `https://api.mux.com/video/v1/assets/${assetId}`,
+          {
+            headers: {
+              Authorization: `Basic ${Buffer.from(
+                `${process.env.MUX_ACCESS_TOKEN_ID}:${process.env.MUX_SECRET_KEY}`,
```
Comment (joelhooks) [inline src/inngest/functions/mux/add-srt-to-mux-asset.ts]: need to check/add all env vars

File: src/inngest/functions/mux/add-srt-to-mux-asset.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+          const trackId = muxAsset.tracks.filter(
+            (track: {type: string; status: string}) => track.type === 'text',
+          )[0]?.id
+          return await fetch(
+            `https://api.mux.com/video/v1/assets/${videoResource.muxAssetId}/tracks/${trackId}`,
+            {
+              method: 'DELETE',
+              headers: {
+                Authorization: `Basic ${Buffer.from(
+                  `${process.env.MUX_ACCESS_TOKEN_ID}:${process.env.MUX_SECRET_KEY}`,
```
Comment (joelhooks) [inline src/inngest/functions/mux/add-srt-to-mux-asset.ts]: need to check/add all env vars

File: src/inngest/functions/mux/add-srt-to-mux-asset.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        })
+
+        await step.run('add srt track to mux asset', async () => {
+          return await fetch(
+            `https://api.mux.com/video/v1/assets/${muxAsset.id}/tracks`,
+            {
+              method: 'POST',
+              headers: {
+                Authorization: `Basic ${Buffer.from(
+                  `${process.env.MUX_ACCESS_TOKEN_ID}:${process.env.MUX_SECRET_KEY}`,
```
Comment (joelhooks) [inline src/inngest/functions/mux/add-srt-to-mux-asset.ts]: need to check/add all env vars

File: src/inngest/functions/mux/mux-webhooks-handlers.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+import {VideoResourceSchema} from 'inngest/functions/transcript-ready'
+
+export const muxVideoAssetCreated = inngest.createFunction(
+  {id: `mux-video-asset-created`, name: 'Mux Video Asset Created'},
+  {
+    event: MUX_WEBHOOK_EVENT,
+    if: 'event.data.muxWebhookEvent.type == "video.asset.created"',
+  },
+  async ({event, step}) => {
+    // TODO Partykit
```
Comment (joelhooks) [inline src/inngest/functions/mux/mux-webhooks-handlers.ts:13]: PK is to "announce" could sub for Slack in here.

File: src/inngest/functions/video-uploaded.ts [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+      const muxOptions = getMuxOptions({
+        url: event.data.originalMediaUrl,
+        passthrough: event.data.fileName,
+      })
+
+      const response = await fetch(`${baseUrl}/video/v1/assets`, {
+        headers: {
+          Authorization: `Basic ${Buffer.from(
+            `${process.env.MUX_ACCESS_TOKEN_ID}:${process.env.MUX_SECRET_KEY}`,
```
Comment (joelhooks) [inline src/inngest/functions/video-uploaded.ts]: need to check/add all env vars (all of the mux ones should be in a lib)

File: src/app/api/inngest/route.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+import {inngest} from 'inngest/inngest.server'
+import {addSrtToMuxAsset} from 'inngest/functions/mux/add-srt-to-mux-asset'
+import {videoUploaded} from 'inngest/functions/video-uploaded'
+import {transcriptReady} from 'inngest/functions/transcript-ready'
+import {
+  muxVideoAssetCreated,
+  muxVideoAssetReady,
+} from 'inngest/functions/mux/mux-webhooks-handlers'
+
+export const {GET, POST, PUT} = serve({
```
Comment (joelhooks) [inline src/app/api/inngest/route.ts:11]: moved inngest over to app router

File: src/utils/sanity.fetch.only.server.ts [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+      headers: {
+        'Content-type': 'application/json',
+        Authorization: `Bearer ${process.env.SANITY_API_TOKEN}`,
+      },
+      body: JSON.stringify({mutations}),
+    },
+  ).then((response) => response.json())
+}
+
+export async function sanityQuery<T = any>(query: string): Promise<T> {
```
Comment (joelhooks) [inline src/utils/sanity.fetch.only.server.ts:15]: these exist to be edge friendly, which is a chore

File: vercel.json [hunk 9]
```diff
@@ -0,0 +1,8 @@
+{
+  "crons": [
+    {
+      "path": "/api/cron",
+      "schedule": "*/5 * * * *"
```
Comment (joelhooks) [inline vercel.json:5]:
```suggestion
      "schedule": "*/1 * * * *"
```

**Review by zacjones93 — APPROVED** [review]

**Pull request merged**